File: Java_Generated/Java/untitled/src/measurements/VectorFormulaConsumer.java
package measurements;

import diagram.interfaces.IDesktop;
import general_service.Entry;
import measurements.interfaces.IStarted;
import measurements.variables.Variable;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class VectorFormulaConsumer extends  DataConsumerMeasurements {
    public VectorFormulaConsumer(String name, IDesktop desktop) {
        super(name, desktop);
    }

    @Override
    public void updateMeasurements() {
      //  mPefrformer.updateChildrenData(this, feedback);
        calculateTree();
        save();
    }

    @Override
    protected void setFeedback(List<Entry<int[], String>> list) {

    }

    @Override
    public void postSetArrow() {
        init();
        createFeedback();
    }

    protected void addVariableValue(String name, Object type)
    {
        var variable = new Variable(type, name, valueSetterFactory);
        addVariable(variable);
    }

    protected void addVariable(Variable variable)
    {
        variables.put(variable.getMeasurementName(), variable);
        addMeasurement(variable);
    }

    protected void save()
    {

    }



    protected void calculateTree() {
    }

protected Map<String, Variable>  variables = new HashMap<>();

}

File: src/Web/OnlineGameConverter/onlinegameconverter.client/Library/Measurements/VectorFormulaConsumer.ts
/* eslint-disable no-var */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-explicit-any */

/* eslint-disable no-var */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-explicit-any */
import type { IDesktop } from "../Interfaces/IDesktop";
import type { IPostSetArrow } from "../Interfaces/IPostSetArrow";
import { DataConsumerVariableMeasurements } from "./DataConsumerVariableMeasurements";

export class VectorFormulaConsumer extends DataConsumerVariableMeasurements implements IPostSetArrow
{

  //  protected arguments: string[] = [];

 //   protected operationNames: Map<number, string> = new Map();

    constructor(desktop: IDesktop, name: string)
    {
        super(desktop, name);
        this.typeName = "VectorFormulaConsumer";
        this.types.push("VectorFormulaConsumer");
        this.types.push("IPostSetArrow");
    }

    updateMeasurements(): void
    {
        if (this.feedback != undefined) {
            this.feedback.setFeedbacks();
        }
        this.calculateTree();
        this.save();
    }

    calculateTree(): void
    {
    }

    init(): void {

    }

    save(): void {

    }



    postSetArrow(): void
    {
        this.init();
        this.setFeedback();
    }


}

